### ISSUE TITLE
Multiple false positive on PH2101: Using 'as' expression

### ISSUE THREAD

**Issue by ynsehoornenborg** [philips-software/roslyn-analyzers#156]
Using version 1.2.29 from NuGet.

The following code throws:
`error PH2101: Using the 'as' expression means 'streamer' could be null.  Either check before dereferencing, or cast if you definitively know it will be this type in this context.`

On the following code
``` cs
var streamer = baseStreamer as FileStreamer;
return streamer != null ? streamer.Toggle(bToggle) : false;
```

If I replace it with this more verbose code, the warning disappears:
``` cs
var streamer = baseStreamer as FileStreamer;
if (streamer != null) {
    streamer.Toggle(bToggle);
}
return false;
```

Preliminary analysis: We need to take ConditionalExpression into account, next to the IfStatement.

**Comment by ynsehoornenborg** [comment]
And another false postive:
```cs
var frameworkElement = sender as FrameworkElement;
if (frameworkElement != null && frameworkElement.IsEnabled)  {
    // bla bla
}
```
Preliminary analysis: IfStatement with multiple conditions is also not taken into account.

**Comment by ynsehoornenborg** [comment]
And the next false positive:
```cs
Date? nullable = value as Date?;
if (nullable.HasValue) {
    // bla bla
}
```

Which fails to notice the `HasValue` property.
